pkg/config/proxy.go

```diff
@@ -528,112 +305,139 @@ func (cfg *HealthCheckConf) checkForCli() error {
 }
 
 // TCP
-type TCPProxyConf struct {
-	BaseProxyConf
-	BindInfoConf
-}
+var _ ProxyConf = &TCPProxyConf{}
 
 func (cfg *TCPProxyConf) Compare(cmp ProxyConf) bool {
 	cmpConf, ok := cmp.(*TCPProxyConf)
 	if !ok {
 		return false
 	}
 
-	if !cfg.BaseProxyConf.compare(&cmpConf.BaseProxyConf) ||
-		!cfg.BindInfoConf.compare(&cmpConf.BindInfoConf) {
+	if !cfg.BaseProxyConf.compare(&cmpConf.BaseProxyConf) {
 		return false
 	}
+
+	// Add custom logic equal if exists.
+	if !reflect.DeepEqual(cfg.TCPProxySpec, cmpConf.TCPProxySpec) {
+		return false
+	}
+
 	return true
 }
 
 func (cfg *TCPProxyConf) UnmarshalFromMsg(pMsg *msg.NewProxy) {
-	cfg.BaseProxyConf.UnmarshalFromMsg(pMsg)
-	cfg.BindInfoConf.UnmarshalFromMsg(pMsg)
+	cfg.BaseProxyConf.unmarshalFromMsg(pMsg)
+
+	// Add custom logic unmarshal if exists
+	cfg.RemotePort = pMsg.RemotePort
 }
 
-func (cfg *TCPProxyConf) UnmarshalFromIni(prefix string, name string, section ini.Section) (err error) {
-	if err = cfg.BaseProxyConf.UnmarshalFromIni(prefix, name, section); err != nil {
-		return
+func (cfg *TCPProxyConf) UnmarshalFromIni(prefix string, name string, section *ini.Section) error {
```

Comment:
```go	
    err := section.MapTo(cfg)
    if err != nil {
        return err
    }

    err = cfg.BaseProxyConf.decorate(prefix, name, section)
    if err != nil {
        return err
    }
```

This code section is repeated in each ProxyConf. Can we wrap it as a function like

`preUnmarshalFromIni(cfg ProxyConf, prefix string,  name string, section *ini.Section)`?

Reply:
I has thinked about it, just like the demo `preUnmarshalFromIni(cfg ProxyConf,...)`, it is complicated when `cfg` is the interface but we want to edit `cfg.BaseProxyConf`, so I just keep it simple.

Reply:
`cfg.GetBaseInfo().decorate(...)`?